rfcs/0000-lifecycle-crd.md

```diff
@@ -0,0 +1,137 @@
+`[Readable](https://github.com/your-name-or-org/kpack/blob/<pr-branch-name>/rfcs/0000-<my-feature>.md)`
+
+**Problem:**
+
+The CNB lifecycle is provided to kpack as in image reference in a configMap similar to how the completion and prepare
+images are passed to the kpack controller. This means that every builder will always use the same exact lifecycle, with
+no ability to specify an alternate one that may support different buildpack api or platform api versions. Due to kpack
+support of windows, a custom lifecycle image with specific layers for the linux and windows binaries is required
+(with a label specifying the layer diff id for each os). Because of this, a cve present in the lifecycle that has been 
+patched upstream, requires any user to build the kpack specific lifecycle image themselves if the kpack provided one
+has not been bumped.
+
+As we [move to deprecate windows](https://github.com/buildpacks-community/kpack/discussions/1366), there is an opportunity for us to utilize
+the upstream lifecycle image shipped by CNB, but switching to that image without changing the interface for providing the
+lifecycle to kpack may open the door to incompatibilities if the wrong image is used on a particular kpack version.
+
+**Outcome:**
+
+Users will be able to create a CR that references a lifecycle image. The file structure in the image must match 
+that of the [CNB Lifecycle image](https://hub.docker.com/r/buildpacksio/lifecycle/tags).
+
+```
+.
+└── cnb/
+    └── lifecycle/
+        ├── analyzer
+        ├── builder
+        ├── detector
+        ├── exporter
+        ├── launcher
+        ├── lifecycle.toml
+        └── restorer
+```
+
+They will be able to use this lifecycle CR in any builder. A lifecycle will be shipped out of the box in kpack and 
+users will not be required to provide a lifecycle reference in the builder if they do not wish to override the default.
+Users should see no difference in functionality compared to the existing kpack lifecycle if they do not choose to 
+provide their own lifecycle CR.
+
+**Actions to take:**
+
+A new ClusterLifecycle CRD will be created. It will be very similar in structure to the ClusterStack Resource. 
+One open question is whether we should follow the Clusterstack model and keep this resource strictly cluster scoped or
+follow the ClusterBuildpack pattern and provide a namespaced scoped version of the CRD. 
```

Comment:
What are the pros and cons here? I could imagine, con of providing additional namespaced scoepd resource is increased complexity. Anything else here?

I'd vote to do whatever is simplest and add more stuff later if it is ever needed or requested.